### PR TITLE
[WIP] support auto-compress for glue task

### PR DESCRIPTION
**Pull request by ceci3**
# What does this PR do?

support auto-compress for glue task

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Hello @ceci3, thank you for your contribution!

Could you just run the code quality tool to ensure that the code quality passes? You can install them with the following, from the root of your clone:
```
pip install -e ".[quality]"
```
And then run them with:
```
make fixup
```

Also, please note that research examples are maintained by the contributor (yourself :) ); so for future PRs that want to update auto-compression, we'll ping you for review. Is that fine with you?

File: examples/research_projects/auto-compression/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+paddle2onnx --model_dir ./tmp/$TASK_NAME/ \
+            --model_filename model.pdmodel \
+            --params_filename model.pdiparams \
+            --save_file model.onnx \
+            --enable_dev_version True
+```
+
+#### 6. Auto Compression
+
+- Single card training:
```
Comment (leiqing1) [inline examples/research_projects/auto-compression/README.md:77]: Train on a Single GPU

File: examples/research_projects/auto-compression/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+```
+export TASK_NAME=mrpc
+
+export CUDA_VISIBLE_DEVICES=0
+python run_glue.py --config_path=./train_config.yaml --task_name=$TASK_NAME --max_length=128 --model_name_or_path=./tmp/$TASK_NAME --per_device_train_batch_size 40 --per_device_eval_batch_size 40 --output_dir ./tmp_ac/$TASK_NAME 
+```
+
+#### 7. Deployment
+
+##### 7.1 Deploy with Paddle TensorRT
```
Comment (leiqing1) [inline examples/research_projects/auto-compression/README.md:87]: 增加7.2：导出ONNX模型的部署。（是提到生态，需要开放。）

File: examples/research_projects/auto-compression/README.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+| model | strategy | CoLA | MRPC | QNLI | QQP | RTE | SST2  | STSB  | AVG |
+|:------:|:------:|:------:|:------:|:-----------:|:------:|:------:|:------:|:------:|:------:|
+| bert-base-cased | origin model | 60.06 | 84.31 | 90.68 | 90.84 | 63.53 | 91.63  | 88.46 |  81.35  |
+| bert-base-cased | structure prune & quant aware | 58.69 | 85.05 | 90.74 | 90.42 | 65.34 | 92.08 | 88.22 |  81.51 |
+
+|  bert-base-cased | Accuracy（avg） | Latency(ms) | SpeedUp |
+|:-------:|:----------:|:------------:| :------:|
+| origin model |  81.35 | - | - |
+| compressed model |  81.51 | - | - |
+
```
Comment (leiqing1) [inline examples/research_projects/auto-compression/README.md:22]: 1. 目前看精度没什么收益。需要确定压缩后速度、显存、内存占用会有收益。
2. compressed model和structure prune & quant aware是一个模型吗？名字保持一致，大家更容易理解。

File: examples/research_projects/auto-compression/README.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  --max_seq_length 128 \
+  --per_device_train_batch_size 32 \
+  --learning_rate 2e-5 \
+  --num_train_epochs 3 \
+  --output_dir ./tmp/$TASK_NAME/
+```
+
+#### 5. Export FP32 model to ONNX
+```
+pip install paddle2onnx
```
Comment (leiqing1) [inline examples/research_projects/auto-compression/README.md:67]: x2paddle

File: examples/research_projects/auto-compression/README.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  --do_train \
+  --do_eval \
+  --max_seq_length 128 \
+  --per_device_train_batch_size 32 \
+  --learning_rate 2e-5 \
+  --num_train_epochs 3 \
+  --output_dir ./tmp/$TASK_NAME/
+```
+
+#### 5. Export FP32 model to ONNX
```
Comment (leiqing1) [inline examples/research_projects/auto-compression/README.md:65]: Exporting FP32 Models for Compression Acceleration

File: examples/research_projects/auto-compression/README.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+| bert-base-cased | origin model | 60.06 | 84.31 | 90.68 | 90.84 | 63.53 | 91.63  | 88.46 |  81.35  |
+| bert-base-cased | structure prune & quant aware | 58.69 | 85.05 | 90.74 | 90.42 | 65.34 | 92.08 | 88.22 |  81.51 |
+
+|  bert-base-cased | Accuracy（avg） | Latency(ms) | SpeedUp |
+|:-------:|:----------:|:------------:| :------:|
+| origin model |  81.35 | - | - |
+| compressed model |  81.51 | - | - |
+
+- Nvidia GPU ：
+  - hardware：NVIDIA Tesla T4 single card
```
Comment (leiqing1) [inline examples/research_projects/auto-compression/README.md:24]: One Nvidia Tesla T4 GPU
